Fenix enrollment test flaky: bare log-state in poll loop cannot recover a failed apply

Follow-up to #15845. After #15846 moved `log-state` into the poll loop, the Fenix Enrollment integration test still fails intermittently — mostly on the `beta` channel:

```
AssertionError: No log-state row found for fenix-<channel>-integration-test after 120s.
```

### Root cause

nimbus-cli's `enroll` launch is self-contained: in one app launch it resets the db, injects the experiment payload, applies it (enrolls), and prints the log-state table (`application-services/components/support/nimbus-cli/src/cmd.rs:367-373`). The recipe is `targeting:"true"` + bucket `count:10000/total:10000` (100%), so enrollment is deterministic — it cannot fail on eligibility.

The poll loop re-issues **bare `log-state`** (`cmd.rs:314-319`), which injects no experiment payload and does not reset. On a loaded beta emulator, the enroll launch's `applyPendingExperiments` doesn't commit to SQLite before the app is backgrounded — the `reset-db` already wiped the db, and the apply never landed. From then on, bare `log-state` has nothing to surface and nothing to recover from, so the loop reports an empty table for the full 120s.

So #15846 re-issued the right cadence but the wrong command.

### Fix

In the poll loop, re-issue the full `enroll` (reset-db + inject payload + apply + log-state) each iteration instead of bare `log-state`, so an incomplete apply self-heals on the next poll. Matches nimbus-cli's design (the enroll launch logs state itself).

### Acceptance criteria

- [ ] The poll loop re-injects the experiment payload each iteration so a late/incomplete apply still surfaces.
- [ ] No change to enrollment authority — Nimbus still owns assignment; this is read-reliability only.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-7011)

--- a/experimenter/tests/integration/nimbus/android/test_fenix_enrollment.py
+++ b/experimenter/tests/integration/nimbus/android/test_fenix_enrollment.py
@@ -8,9 +8,8 @@ import pytest
 from nimbus.models.base_dataclass import BaseExperimentApplications
 
 FENIX_APP = BaseExperimentApplications.FIREFOX_FENIX.value
-APP_APPLY_WAIT = 15
-LOG_STATE_TIMEOUT = 120
-LOG_STATE_POLL_INTERVAL = 5
+ENROLL_TIMEOUT = 120
+ENROLL_POLL_INTERVAL = 5
 
 
 @pytest.mark.fenix_enrollment
@@ -27,7 +26,6 @@ def test_fenix_enrollment(
     recipe_path.write_text(json.dumps(recipe))
 
     subprocess.check_call(["adb", "install", fenix_apk_path])
-    subprocess.check_call(["adb", "logcat", "-c"])
 
     # Prevent the real Nimbus RS fetch from overwriting our local enrollment
     # (without this, Nimbus evolves against the fresh fetch — which does not
@@ -36,26 +34,26 @@ def test_fenix_enrollment(
         ["adb", "shell", "cmd", "connectivity", "airplane-mode", "enable"]
     )
 
-    subprocess.check_call(
-        [
-            "nimbus-cli",
-            "--app",
-            FENIX_APP,
-            "--channel",
-            fenix_channel,
-            "enroll",
-            experiment_slug,
-            "--branch",
-            "control",
-            "--file",
-            str(recipe_path),
-            "--preserve-targeting",
-            "--preserve-bucketing",
-            "--reset-app",
-            "--no-validate",
-        ]
-    )
-    time.sleep(APP_APPLY_WAIT)
+    def enroll():
+        subprocess.check_call(
+            [
+                "nimbus-cli",
+                "--app",
+                FENIX_APP,
+                "--channel",
+                fenix_channel,
+                "enroll",
+                experiment_slug,
+                "--branch",
+                "control",
+                "--file",
+                str(recipe_path),
+                "--preserve-targeting",
+                "--preserve-bucketing",
+                "--reset-app",
+                "--no-validate",
+            ]
+        )
 
     pattern = re.compile(
         rf"nimbus_client:\s*{re.escape(experiment_slug)}\s+\|\s*\S+\s+\|\s*(\S+)"
@@ -63,13 +61,11 @@ def test_fenix_enrollment(
 
     logcat = ""
     match = None
-    deadline = time.monotonic() + LOG_STATE_TIMEOUT
+    deadline = time.monotonic() + ENROLL_TIMEOUT
     while time.monotonic() < deadline:
         subprocess.check_call(["adb", "logcat", "-c"])
-        subprocess.check_call(
-            ["nimbus-cli", "--app", FENIX_APP, "--channel", fenix_channel, "log-state"]
-        )
-        time.sleep(LOG_STATE_POLL_INTERVAL)
+        enroll()
+        time.sleep(ENROLL_POLL_INTERVAL)
         logcat = subprocess.check_output(["adb", "logcat", "-d"], text=True)
         match = pattern.search(logcat)
         if match is not None:
@@ -77,7 +73,7 @@ def test_fenix_enrollment(
 
     nimbus_lines = [line for line in logcat.splitlines() if "nimbus_client" in line]
     assert match is not None, (
-        f"No log-state row found for {experiment_slug} after {LOG_STATE_TIMEOUT}s.\n"
+        f"No enrollment row found for {experiment_slug} after {ENROLL_TIMEOUT}s.\n"
         f"--- last 30 nimbus_client lines ---\n" + "\n".join(nimbus_lines[-30:])
     )
     enrolled_branch = match.group(1)

